appintro/src/main/res/layout/appintro_intro_layout2.xml

```diff
@@ -1,96 +1,80 @@
-                style="@style/AppIntroButtonStyleCompat"
-                android:layout_width="wrap_content"
-                android:layout_height="wrap_content"
-                android:layout_gravity="start"
-                android:layout_marginLeft="16dp"
-                android:layout_marginStart="16dp"
-                android:visibility="invisible"
-                android:contentDescription="@string/app_intro_back_button"
-                app:srcCompat="@drawable/ic_appintro_arrow_back_white" />
-
-            <ImageButton
-                android:id="@+id/next"
-                style="@style/AppIntroButtonStyleCompat"
-                android:layout_width="wrap_content"
-                android:layout_height="wrap_content"
-                android:layout_gravity="end"
-                android:layout_marginEnd="16dp"
-                android:layout_marginRight="16dp"
-                android:contentDescription="@string/app_intro_next_button"
-                app:srcCompat="@drawable/ic_appintro_arrow_forward_white" />
-
-            <ImageButton
-                android:id="@+id/done"
-                style="@style/AppIntroButtonStyleCompat"
-                android:layout_width="wrap_content"
-                android:layout_height="wrap_content"
-                android:layout_gravity="end"
-                android:layout_marginEnd="16dp"
-                android:layout_marginRight="16dp"
-                android:visibility="gone"
-                android:contentDescription="@string/app_intro_done_button"
-                app:srcCompat="@drawable/ic_appintro_done_white" />
-        </FrameLayout>
-    </LinearLayout>
-</RelativeLayout>
+<?xml version="1.0" encoding="utf-8"?>
+<androidx.constraintlayout.widget.ConstraintLayout xmlns:android="http://schemas.android.com/apk/res/android"
+    xmlns:app="http://schemas.android.com/apk/res-auto"
+    xmlns:tools="http://schemas.android.com/tools"
+    android:id="@+id/background"
+    android:layout_width="match_parent"
+    android:layout_height="match_parent"
+    android:background="@color/appintro_background_color"
+    android:layoutDirection="ltr"
+    tools:theme="@style/Theme.AppCompat.Light.NoActionBar">
+
+    <com.github.paolorotolo.appintro.AppIntroViewPager
+        android:id="@+id/view_pager"
+        android:layout_width="0dp"
+        android:layout_height="0dp"
+        android:fitsSystemWindows="true"
+        android:overScrollMode="never"
+        app:layout_constraintBottom_toTopOf="@+id/bottom"
+        app:layout_constraintEnd_toEndOf="parent"
+        app:layout_constraintStart_toStartOf="parent"
+        app:layout_constraintTop_toTopOf="parent" />
+
+    <!--This FrameLayout should not be removed and must always have a transparent background if a custom background is set to -->
+    <!--@+id/background, otherwise the ripple effect on the buttons will not appear-->
+    <!--See https://stackoverflow.com/questions/31948189/ for more info-->
+    <FrameLayout

```

Comment:
Okay. I will see and report back

Reply:
@cortinico So I tried using the foreground and the result it produces is a square ripple instead of circle, so instead of setting foreground, I used background, removed the framelayout and change the method in appintro2 to set the background image and not an imageview

Reply:
Ok that should do the job also :)